packages/daemon/src/types.d.ts

```diff
@@ -267,3 +218,63 @@ export type EndoWebBundle = {
   bundle: ERef<EndoReadable>;
   powers: ERef<unknown>;
 };
+
+export type DiskPowers = {
+  makeFileReader: (path: string) => Reader<Uint8Array>;
+  makeFileWriter: (path: string) => Writer<Uint8Array>;
+  writeFileText: (path: string, text: string) => Promise<void>;
+  readFileText: (path: string) => Promise<string>;
+  maybeReadFileText: (path: string) => Promise<string | undefined>;
+  readDirectory: (path: string) => Promise<Array<string>>;
+  makePath: (path: string) => Promise<void>;
+  joinPath: (...components: Array<string>) => string;
+  removePath: (path: string) => Promise<void>;
+  renamePath: (source: string, target: string) => Promise<void>;
+};
+
+export type DaemonicPowers = {
+  env: Record<string, string | undefined>;
+  sinkError: (error) => void;
+  makeSha512: () => Sha512;
+  randomHex512: () => Promise<string>;
+  servePath: (args: {
+    path: string;
+    host?: string;
+    cancelled: Promise<never>;
+  }) => Promise<AsyncIterableIterator<Connection>>;
+  servePort: (args: {
+    port: number;
+    host?: string;
+    cancelled: Promise<never>;
+  }) => Promise<AsyncIterableIterator<Connection>>;
+  servePortHttp: (args: {
+    port: number;
+    host?: string;
+    respond?: HttpRespond;
+    connect?: HttpConnect;
+    cancelled: Promise<never>;
+  }) => Promise<number>;
+  informParentWhenReady: () => void;
+  reportErrorToParent: (message: string) => void;
+  delay: (ms: number, cancelled: Promise<never>) => Promise<void>;
+  makeWorker: (
+    id: string,
+    locator: Locator,
+    cancelled: Promise<never>,
+  ) => Promise<Worker>;
+  fileURLToPath: (url: string) => string;
+  makeHashedContentWriter: (statePath: string) => Promise<{
+    writer: Writer<Uint8Array>;
+    getSha512Hex: () => Promise<string>;
+  }>;
+  makeHashedContentReadeableBlob: (statePath: string, sha512: string) => {
+    stream: () => Promise<FarRef<Reader<Uint8Array>>>;
```

Comment:
is `FarRef` correct?

Reply:
I think `Promise<FarRef<T>>` simplifies to `FarRef<T>` and it seems right to me. cc @michaelfig

Reply:
If you intend for `stream` to be potentially remote but returned `async`, then you indeed need `Promise<FarRef<T>>`.  If it's potentially remote but synchronously returned, you can get away with `FarRef<T>`.